src/mrinufft/operators/off_resonance.py

```diff
@@ -0,0 +1,325 @@
+"""Off Resonance correction Operator wrapper.
+
+Based on the implementation of Guillaume Daval-Frérot in pysap-mri:
+https://github.com/CEA-COSMIC/pysap-mri/blob/master/mri/operators/fourier/orc_wrapper.py
+"""
+
+import math
+import numpy as np
+
+from .._utils import get_array_module
+
+from .base import FourierOperatorBase
+from .interfaces.utils import is_cuda_array
+
+CUPY_AVAILABLE = True
+try:
+    import cupy as cp
+except ImportError:
+    CUPY_AVAILABLE = False
+
+
+TORCH_AVAILABLE = True
+try:
+    import torch
+except ImportError:
+    TORCH_AVAILABLE = False
+
+
+def get_interpolators_from_fieldmap(
+    fieldmap, t, n_time_segments=6, n_bins=(40, 10), mask=None
+):
+    r"""Create B and C matrices to approximate ``exp(-2j*pi*fieldmap*t)``.
+
+    Here, B has shape ``(n_time_segments, len(t))``
+    and C has shape ``(n_time_segments, *fieldmap.shape)``.
+
+    From Sigpy: https://github.com/mikgroup/sigpy
+    and MIRT (mri_exp_approx.m): https://web.eecs.umich.edu/~fessler/code/
+
+    Parameters
+    ----------
+    fieldmap : np.ndarray or GPUarray
```

Comment:
Do we use GPUArray @paquiteau. Just want to ensure documentation is streamlined.

Reply:
What about using npt.ArrayLike?

Reply:
I would say that `ndarray` is good enough for a docstring, with a comment saying that this compatible with cupy and torch.